app/controllers/payments/WaysToPayController.scala

```diff
@@ -83,12 +82,12 @@ class WaysToPayController @Inject()(
 
   def progressToPayment(fn: (FeeResponse, String, String) => Future[Result])
                        (errorMessage: String)
-                       (implicit ac: AuthContext, hc: HeaderCarrier): Future[Result] = {
+                       (implicit hc: HeaderCarrier, request:AuthorisedRequest[_]): Future[Result] = {
 
     val submissionDetails = for {
-      amlsRefNo <- OptionT(authEnrolmentsService.amlsRegistrationNumber)
-      (_, detailedStatus) <- OptionT.liftF(statusService.getDetailedStatus(amlsRefNo))
-      fees <- OptionT(feeResponseService.getFeeResponse(amlsRefNo))
+      amlsRefNo <- OptionT(authEnrolmentsService.amlsRegistrationNumber(request.amlsRefNumber, request.groupIdentifier))
```

Comment:
As above; we should not use this method. Get it from the authorised request which retrieves the enrolments up front.

Reply:
Using the updated version of authEnrolmentsService.amlsRegistrationNumber by @slavvco, as it is being mocked in tests